### DISCUSSION TITLE
I am trying to convert point cloud to mesh using Openvdb getting two-sided mesh

### DISCUSSION THREAD

**Discussion by sonamkumar123** [AcademySoftwareFoundation/openvdb#1265] [Q&A]
I am trying to convert point cloud to mesh using Openvdb getting two-sided mesh.
Using code – 

const float voxelSize = 0.50f, halfWidth = 2.0f;
openvdb::FloatGrid::Ptr grid = openvdb::createLevelSet<openvdb::FloatGrid>(voxelSize, halfWidth);
openvdb::tools::ParticlesToLevelSet<openvdb::FloatGrid> raster(*grid);

raster.setGrainSize(1);//a value of zero disables threading
raster.rasterizeSpheres(pa, 1);

std::vector<openvdb::Vec3s> points;
std::vector<openvdb::Vec4I> quads;
std::vector<openvdb::Vec3I> triangles;

openvdb::tools::volumeToMesh(*grid, points, triangles, quads, 1, 0);

Trying to create a mesh using torus point cloud, getting two-sided mesh(torus inside torus) as output. [Outputimage](https://drive.google.com/file/d/1bUCQO1-e-42kh7ATNJvw1i98Ii69yItv/view?usp=sharing)

**Comment by Idclip** [comment]
Hi @sonamkumar123 - try running an single invocation of `tools::levelSetRebuild()` after `tools::volumeToMesh` and see if that fixes your issue
